src/crimson/os/seastore/transaction_manager.h

```diff
@@ -1031,36 +1002,34 @@ class TransactionManager : public ExtentCallbackInterface {
   template <typename T>
   pin_to_extent_ret<T> pin_to_extent(
     Transaction &t,
-    LBAMappingRef pin,
+    LBAMapping pin,
     extent_len_t direct_partial_off,
     extent_len_t partial_len) {
     static_assert(is_logical_type(T::TYPE));
     using ret = pin_to_extent_ret<T>;
-    auto &pref = *pin;
-    auto direct_length = pref.is_indirect() ?
-      pref.get_intermediate_length() :
-      pref.get_length();
+    auto direct_length = pin.is_indirect() ?
+      pin.get_intermediate_length() :
+      pin.get_length();
```

Comment:
It should be safe to reduce the above 3 lines into `auto direct_length = pin.get_intermediate_length()`, and drop the internal `assert(is_indirect());`.